src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -588,9 +635,18 @@ private int decodeOpen(
 
         if (open != null)
         {
-            server.onDecodeOpen(traceId, authorization, open);
-            server.decoder = decodeFrameType;
-            progress = open.limit();
+            if (open.sizeof() + FRAME_HEADER_SIZE > MIN_MAX_FRAME_SIZE)
+            {
+                server.onDecodeError(traceId, authorization, CONNECTION_FRAMING_ERROR);
+                server.decoder = decodeFrameType;
+                progress = limit;
+            }
```

Comment:
This check should be removed, as it will be covered by initializing the `AmqpServer` member field `decodeMaxFrameSize` to `MIN_MAX_FRAME_SIZE`.